hub/util/auto.py

```diff
@@ -0,0 +1,41 @@
+import glob
+import os, random
+from collections import Counter
+
+
+def detect_compression(path: str):
+    """Detects the most frequent compression type of files in a dataset.
+
+    Args:
+        path (str): Path to the directory of a local dataset.
+
+    Returns:
+        compression (str): Compression type of a dataset.
+    """
+
+    types = (".jpeg", ".png", ".jpg")
```

Comment:
you should make `types` a parameter to this function, and make this value the default.

also, please rename this argument to `allowed_extensions`